app/api/v2/views/offices.py

```diff
@@ -75,3 +77,41 @@ def get_specific_office(office_id):
     if office:
         return utils.response_fn(200, "data", office)
     return utils.response_fn(404, "error", "Office is not not found")
+
+
+@path_2.route("/offices/<int:office_id>/register", methods=["POST"])
+@token_required
+def register_candidate_to_office(userobj, office_id):
+    """
+    this is where we check if the candidates information is
+    eligible so that it can be registered to an office.
+    """
+    try:
+        email = userobj[0][0]
+    except:
+        return utils.response_fn(401, "error", "You don't have an account")
+
+    try:
+        data = request.get_json()
+        user = data["user"]
+
+    except KeyError:
+        abort(utils.response_fn(400, "error", "Keys Should be office & user"))
+
+    # check if details are for an admin.
+    if email == "tevinthuku@gmail.com":
+        # does the candidate & office exist in the db.
+        candidate = UserModel.get_user_by_id(user)
+        office = OfficesModel.get_specific_office(office_id)
+        if candidate and office:
+            is_candidate_registered = CandidateModel.check_if_candidate_is_already_registered(
+                candidate[0][0], office[0]["id"])
+            if is_candidate_registered:
+                return utils.response_fn(400, "error", "Candidate is already registerd in this office")
+            CandidateModel.register_politician_user_to_office(
+                office[0]["id"], candidate[0][0])
+            return utils.response_fn(201, "message", "registered candidate")
+        else:
+            return utils.response_fn(404, "error", "Either candidate or office is missing in db")
+
+    return utils.response_fn(401, "error", "You are not authorized.")
```

Comment:
Avoid too many `return` statements within this function, it will affect your maintainability. You can abstract parts of it.

Reply:
I have now implemented a function that abstracts the isAdmin check. It aborts with 401 when email does not match that of the user.